DAOS-623 build: Remove Quick-build option and support.

ashleypittman opened the pull request:
Improved use of Docker caching layers mean this flag does not
actually improve the build times at all so it adds complexity for
little benefit.  Remove support for Quickbuild from Jenkinsfile,
setup scripts and docker files.

With this PR then using the Quick-build pragma will skip check and NLT
stages still - this is handled through pipeline-lib.

daosbuild1 commented:
Test stage Unit Test bdev with memcheck on EL 8 completed with status FAILURE.  https://build.hpdd.intel.com/job/daos-stack/job/daos/job/PR-13004/7/display/redirect

daosbuild1 commented:
Test stage Unit Test bdev on EL 8 completed with status FAILURE.  https://build.hpdd.intel.com/job/daos-stack/job/daos/job/PR-13004/7/display/redirect

daosbuild1 commented:
Test stage Unit Test with memcheck on EL 8 completed with status FAILURE.  https://build.hpdd.intel.com/job/daos-stack/job/daos/job/PR-13004/7/display/redirect

daosbuild1 commented:
Test stage Unit Test on EL 8 completed with status FAILURE.  https://build.hpdd.intel.com/job/daos-stack/job/daos/job/PR-13004/7/display/redirect

daosbuild1 commented:
Test stage Build RPM on EL 9 completed with status FAILURE.  https://build.hpdd.intel.com//job/daos-stack/job/daos/view/change-requests/job/PR-13004/8/execution/node/238/log

daosbuild1 commented:
Test stage Build RPM on EL 8 completed with status FAILURE.  https://build.hpdd.intel.com//job/daos-stack/job/daos/view/change-requests/job/PR-13004/8/execution/node/232/log

daosbuild1 commented:
Test stage Build RPM on Leap 15.4 completed with status FAILURE.  https://build.hpdd.intel.com//job/daos-stack/job/daos/view/change-requests/job/PR-13004/8/execution/node/245/log

daosbuild1 commented:
Test stage Build DEB on Ubuntu 20.04 completed with status FAILURE.  https://build.hpdd.intel.com//job/daos-stack/job/daos/view/change-requests/job/PR-13004/8/execution/node/277/log

daosbuild1 commented:
Test stage Build RPM on EL 9 completed with status FAILURE.  https://build.hpdd.intel.com//job/daos-stack/job/daos/view/change-requests/job/PR-13004/9/execution/node/267/log

daosbuild1 commented:
Test stage Build RPM on EL 8 completed with status FAILURE.  https://build.hpdd.intel.com//job/daos-stack/job/daos/view/change-requests/job/PR-13004/9/execution/node/262/log

daosbuild1 commented:
Test stage Build RPM on Leap 15.4 completed with status FAILURE.  https://build.hpdd.intel.com//job/daos-stack/job/daos/view/change-requests/job/PR-13004/9/execution/node/263/log

daosbuild1 commented:
Test stage Build DEB on Ubuntu 20.04 completed with status FAILURE.  https://build.hpdd.intel.com//job/daos-stack/job/daos/view/change-requests/job/PR-13004/9/execution/node/276/log

daosbuild1 commented on utils/rpms/packaging/rpm_chrootbuild:
```diff
@@ -117,11 +117,7 @@ echo "\"\"\"" >> "$cfg_file"
-    rpm -q mock
-    mock --debug-config
-fi
+eval mock --configdir "$mock_config_dir" -r "${CHROOT_NAME}" ${repo_dels[*]} ${repo_adds[*]}       \
```
(lint) Use "${array[@]}" (with quotes) to prevent whitespace problems. [SC2048]
(lint) Use "${array[@]}" (with quotes) to prevent whitespace problems. [SC2048]

daosbuild1 commented on utils/rpms/packaging/rpm_chrootbuild:
```diff
@@ -117,11 +117,7 @@
-    mock --debug-config
-fi
+eval mock --configdir "$mock_config_dir" -r "${CHROOT_NAME}" ${repo_dels[*]} ${repo_adds[*]}       \
+    --disablerepo=\*-debug* "${releasever_opt[@]}" $MOCK_OPTIONS $RPM_BUILD_OPTIONS "$TARGET"
```
(lint) eval negates the benefit of arrays. Drop eval to preserve whitespace/symbols (or eval as string). [SC2294]

daosbuild1 reviewed (changes requested):
Style warning(s) for job https://build.hpdd.intel.com/job/daos-stack/job/daos/job/PR-13004/10/
Please review https://wiki.hpdd.intel.com/display/DC/Coding+Rules

FYI: Errors found in lines not modified in the patch:

[ci/rpm/build_unsuccessful.sh:12](https://github.com/daos-stack/daos/blob/66c1496bdf633e160ddfe4a48ad21082be62d659/ci/rpm/build_unsuccessful.sh#L12):
(lint) ShellCheck can't follow non-constant source. Use a directive to specify location. [SC1090]

daosbuild1 commented:
Test stage Build RPM on EL 9 completed with status FAILURE.  https://build.hpdd.intel.com//job/daos-stack/job/daos/view/change-requests/job/PR-13004/10/execution/node/378/log

daosbuild1 commented:
Test stage Build RPM on EL 8 completed with status FAILURE.  https://build.hpdd.intel.com//job/daos-stack/job/daos/view/change-requests/job/PR-13004/10/execution/node/366/log

daosbuild1 commented:
Test stage Build RPM on Leap 15.4 completed with status FAILURE.  https://build.hpdd.intel.com//job/daos-stack/job/daos/view/change-requests/job/PR-13004/10/execution/node/349/log

daosbuild1 commented:
Test stage Build DEB on Ubuntu 20.04 completed with status FAILURE.  https://build.hpdd.intel.com//job/daos-stack/job/daos/view/change-requests/job/PR-13004/10/execution/node/303/log

daosbuild1 commented:
Test stage Unit Test bdev with memcheck on EL 8 completed with status FAILURE.  https://build.hpdd.intel.com/job/daos-stack/job/daos/job/PR-13004/10/display/redirect

daosbuild1 commented:
Test stage Unit Test bdev on EL 8 completed with status FAILURE.  https://build.hpdd.intel.com/job/daos-stack/job/daos/job/PR-13004/10/display/redirect

daosbuild1 commented:
Test stage Unit Test on EL 8 completed with status FAILURE.  https://build.hpdd.intel.com/job/daos-stack/job/daos/job/PR-13004/10/display/redirect

daosbuild1 commented:
Test stage Unit Test with memcheck on EL 8 completed with status FAILURE.  https://build.hpdd.intel.com/job/daos-stack/job/daos/job/PR-13004/10/display/redirect

daosbuild1 commented:
Test stage NLT on EL 8 completed with status FAILURE.  https://build.hpdd.intel.com/job/daos-stack/job/daos/job/PR-13004/10/display/redirect

daosbuild1 reviewed (commented):
LGTM.  No errors found by checkpatch.

daosbuild1 commented:
Test stage Build RPM on EL 8 completed with status FAILURE.  https://build.hpdd.intel.com//job/daos-stack/job/daos/view/change-requests/job/PR-13004/11/execution/node/303/log

daosbuild1 commented:
Test stage Build RPM on EL 9 completed with status FAILURE.  https://build.hpdd.intel.com//job/daos-stack/job/daos/view/change-requests/job/PR-13004/11/execution/node/335/log

daosbuild1 commented:
Test stage Build RPM on Leap 15.4 completed with status FAILURE.  https://build.hpdd.intel.com//job/daos-stack/job/daos/view/change-requests/job/PR-13004/11/execution/node/453/log

daosbuild1 commented:
Test stage Build DEB on Ubuntu 20.04 completed with status FAILURE.  https://build.hpdd.intel.com//job/daos-stack/job/daos/view/change-requests/job/PR-13004/11/execution/node/338/log

daosbuild1 commented:
Test stage Unit Test bdev with memcheck on EL 8 completed with status FAILURE.  https://build.hpdd.intel.com/job/daos-stack/job/daos/job/PR-13004/11/display/redirect

daosbuild1 commented:
Test stage Unit Test bdev on EL 8 completed with status FAILURE.  https://build.hpdd.intel.com/job/daos-stack/job/daos/job/PR-13004/11/display/redirect

daosbuild1 commented:
Test stage Unit Test on EL 8 completed with status FAILURE.  https://build.hpdd.intel.com/job/daos-stack/job/daos/job/PR-13004/11/display/redirect

daosbuild1 commented:
Test stage Unit Test with memcheck on EL 8 completed with status FAILURE.  https://build.hpdd.intel.com/job/daos-stack/job/daos/job/PR-13004/11/display/redirect

daosbuild1 reviewed (commented):
LGTM.  No errors found by checkpatch.

daosbuild1 commented:
Test stage Build RPM on EL 9 completed with status FAILURE.  https://build.hpdd.intel.com//job/daos-stack/job/daos/view/change-requests/job/PR-13004/12/execution/node/406/log

daosbuild1 commented:
Test stage Build RPM on EL 8 completed with status FAILURE.  https://build.hpdd.intel.com//job/daos-stack/job/daos/view/change-requests/job/PR-13004/12/execution/node/405/log

daosbuild1 commented:
Test stage Build RPM on Leap 15.4 completed with status FAILURE.  https://build.hpdd.intel.com//job/daos-stack/job/daos/view/change-requests/job/PR-13004/12/execution/node/349/log

daosbuild1 commented:
Test stage Build DEB on Ubuntu 20.04 completed with status FAILURE.  https://build.hpdd.intel.com//job/daos-stack/job/daos/view/change-requests/job/PR-13004/12/execution/node/380/log

daosbuild1 commented:
Test stage Unit Test bdev on EL 8 completed with status FAILURE.  https://build.hpdd.intel.com/job/daos-stack/job/daos/job/PR-13004/12/display/redirect

daosbuild1 commented:
Test stage Unit Test on EL 8 completed with status FAILURE.  https://build.hpdd.intel.com/job/daos-stack/job/daos/job/PR-13004/12/display/redirect

daosbuild1 commented:
Test stage Unit Test with memcheck on EL 8 completed with status FAILURE.  https://build.hpdd.intel.com/job/daos-stack/job/daos/job/PR-13004/12/display/redirect

daosbuild1 commented:
Test stage NLT on EL 8 completed with status FAILURE.  https://build.hpdd.intel.com/job/daos-stack/job/daos/job/PR-13004/12/display/redirect

daosbuild1 reviewed (commented):
LGTM.  No errors found by checkpatch.

daosbuild1 commented:
Test stage Build RPM on EL 9 completed with status FAILURE.  https://build.hpdd.intel.com//job/daos-stack/job/daos/view/change-requests/job/PR-13004/13/execution/node/294/log

daosbuild1 commented:
Test stage Build RPM on Leap 15.4 completed with status FAILURE.  https://build.hpdd.intel.com//job/daos-stack/job/daos/view/change-requests/job/PR-13004/13/execution/node/366/log

daosbuild1 commented:
Test stage Build DEB on Ubuntu 20.04 completed with status FAILURE.  https://build.hpdd.intel.com//job/daos-stack/job/daos/view/change-requests/job/PR-13004/13/execution/node/363/log

daosbuild1 commented:
Test stage Build RPM on EL 8 completed with status FAILURE.  https://build.hpdd.intel.com//job/daos-stack/job/daos/view/change-requests/job/PR-13004/13/execution/node/419/log

brianjmurrell reviewed (changes requested):
As discussed yesterday, let's please isolate the QB removal from the other parts of this PR so that any potential need to revert the QB removal will not inadvertently remove the other desired bits.

ashleypittman commented:
> As discussed yesterday, let's please isolate the QB removal from the other parts of this PR so that any potential need to revert the QB removal will not inadvertently remove the other desired bits.

Done.  We now need to look at https://github.com/daos-stack/daos/pull/12941 and https://github.com/daos-stack/daos/pull/13014 in addition to this PR.

daosbuild1 commented:
Test stage Build RPM on EL 8 completed with status FAILURE.  https://build.hpdd.intel.com/job/daos-stack/job/daos/job/PR-13004/14/display/redirect

JohnMalmberg commented:
Bullseye builds used to be dependent on the quickbuild pragma.  I think that changed when the Dockerfiles were updated to use cached images.   It may be good to run build with Bullseye unit tests enabled.

In the past changes to quickbuild tended to break the Bullseye build stage.  Added @dinghwah as a reviewer.

daosbuild1 reviewed (commented):
LGTM.  No errors found by checkpatch.

ashleypittman commented:
> Bullseye builds used to be dependent on the quickbuild pragma. I think that changed when the Dockerfiles were updated to use cached images. It may be good to run build with Bullseye unit tests enabled.
> 
> In the past changes to quickbuild tended to break the Bullseye build stage. Added @dinghwah as a reviewer.

I've updated to try the bullseye stages.

daosbuild1 reviewed (commented):
LGTM.  No errors found by checkpatch.

daosbuild1 commented:
Test stage Build RPM on EL 9 completed with status FAILURE.  https://build.hpdd.intel.com//job/daos-stack/job/daos/view/change-requests/job/PR-13004/17/execution/node/346/log

daosbuild1 commented:
Test stage Build RPM on EL 8 completed with status FAILURE.  https://build.hpdd.intel.com//job/daos-stack/job/daos/view/change-requests/job/PR-13004/17/execution/node/335/log

daosbuild1 reviewed (commented):
LGTM.  No errors found by checkpatch.

brianjmurrell commented:
@ashleypittman In https://build.hpdd.intel.com/blue/organizations/jenkins/daos-stack%2Fdaos/detail/PR-13004/18/pipeline/216 the EL8 non-RPM build took only a few minutes (so pressumably nothing changed that needed the dependency stack rebuilt in docker) but the leap 15/icc build took over 18 minutes and did rebuild the dependency stack in docker.

Do we understand why?  Shouldn't the conditions for dependency rebuild be the same for both of those?

brianjmurrell commented on ci/unit/required_packages.sh:
```diff
@@ -37,23 +38,6 @@ pkgs="argobots                         \
-        cat "$distro"-required-mercury-rpm-version
-        exit 1
-    fi
-    pkgs+=" spdk-tools mercury\ \>=\ $mercury_version"
```
If you don't need this mercury then we don't need the `"$distro"-required-mercury-rpm-version` file and it's creation in https://github.com/daos-stack/daos/blob/c7ad81d79458344b8fc1b69ab2b4af0da840ca1b/ci/rpm/build_success.sh#L47C1-L48 should be removed also.

ashleypittman replied:
I've a feeling I tried that before and it's required by pipeline-lib which is frustrating.  Do you mind if I look at this seperately?

ashleypittman replied:
I created https://github.com/daos-stack/daos/pull/13168 which is this PR with those additional two lines removed and as I thought it failed to build with errors from pipeline-lib so I think we might be stuck with this.

ashleypittman replied:
On the other hand I could change that code to just create the file and add a comment as to why it's needed, I'd rather do that in a follow-up PR however.

ashleypittman replied:
I was trying to make progress on this without modifying `pipeline-lib`.

ashleypittman replied:
Awesome.  I'll update this PR tomorrow.

ashleypittman replied:
Done.  This PR now passes without that, the `pipeline-lib` PR will need to land and then I'll have to update this again to remove the branch reference however.

ashleypittman commented:
> @ashleypittman In https://build.hpdd.intel.com/blue/organizations/jenkins/daos-stack%2Fdaos/detail/PR-13004/18/pipeline/216 the EL8 non-RPM build took only a few minutes (so pressumably nothing changed that needed the dependency stack rebuilt in docker) but the leap 15/icc build took over 18 minutes and did rebuild the dependency stack in docker.
> 
> Do we understand why? Shouldn't the conditions for dependency rebuild be the same for both of those?

Yes, in theory that's correct.  This PR does make some changes to the dockerfile so will not share a cache with any other PR and it was also about three weeks since it last ran, plus the fact there's two docker hosts so it's 50:50 as to if the cache would have been populated anyway.  I'm more surprised given that the el8 build was a cache hit than I am that the leap build was a cache miss.

daosbuild1 reviewed (commented):
LGTM.  No errors found by checkpatch.

daosbuild1 reviewed (commented):
LGTM.  No errors found by checkpatch.

daosbuild1 reviewed (commented):
LGTM.  No errors found by checkpatch.

FYI: Errors found in lines not modified in the patch:

[ci/rpm/build_success.sh:12](https://github.com/daos-stack/daos/blob/faf60e31fcd3c0f014becca6d5c55851e7b61ac2/ci/rpm/build_success.sh#L12):
(lint) ShellCheck can't follow non-constant source. Use a directive to specify location. [SC1090]

daosbuild1 reviewed (commented):
LGTM.  No errors found by checkpatch.

brianjmurrell reviewed (dismissed):
Apart from the pipeline-lib reference, +1.

ashleypittman commented on Jenkinsfile:
```diff
@@ -17,6 +17,8 @@
 //@Library(value='pipeline-lib@your_branch') _
 
+@Library(value='pipeline-lib@bmurrell/only-stash-mercury-version-if-exists') _
+
```
Done.

ashleypittman commented:
Refreshed and updated, this should be ready to land now.

daosbuild1 reviewed (commented):
LGTM.  No errors found by checkpatch.

brianjmurrell commented:
I know it's not part of your PR, but does the Codespell failure need fixing?  Didn't I see discussion from last week of making that _required_?

Once again, the latest run on Jenkins, the EL8 build took less than 2 minutes but the Leap/Intel build took almost 15 minutes.  :-(  Do we know why?

ashleypittman commented:
> I know it's not part of your PR, but does the Codespell failure need fixing? Didn't I see discussion from last week of making that _required_?

Yes I know.  I was out last week but it seems we landed yet another regression.  We need to land a fix before we can make the step required.

> Once again, the latest run on Jenkins, the EL8 build took less than 2 minutes but the Leap/Intel build took almost 15 minutes. :-( Do we know why?

daosbuild1 reviewed (commented):
LGTM.  No errors found by checkpatch.
